### DISCUSSION TITLE
Dotenv Env

### DISCUSSION THREAD

**Discussion by Holymaiden** [obytes/react-native-template-obytes#297] [Q&A] [answered]
How can I set up dotenv to be used in env.js? I've repeatedly created new projects but when I set up dotenv, I always encounter errors like fs, os, or others.

**Comment by yjose** [comment, marked as answer]
`dotenv` can't be used in React Native during runtime and should only be used at build time. This is precisely what we are doing with the project structure. If you check carefully, we are importing the `env.js` only in `app.config.ts`, as this will be used by Expo while generating the project (build time). To use those environment variables in the `src` project(runtime), we pass all those environment variables as extras in `app.config.ts` and then export them in `src/core/env.js`. make sure to read the [guide for more details](https://starter.obytes.com/getting-started/environment-vars-config/#how-it-works)
